plugins/repository-gcs/src/test/java/io/crate/gcs/GCSBlobStoreContainerTests.java

```diff
@@ -0,0 +1,105 @@
+import com.google.cloud.storage.StorageBatchResult;
+import com.google.cloud.storage.StorageException;
+
+public class GCSBlobStoreContainerTests extends IntegTestCase {
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testDeleteBlobsIgnoringIfNotExistsThrowsIOException() throws Exception {
+        final List<String> blobs = Arrays.asList("blobA", "blobB");
+
+        final StorageBatch batch = mock(StorageBatch.class);
+        if (randomBoolean()) {
+            StorageBatchResult<Boolean> result = mock(StorageBatchResult.class);
+            when(batch.delete(any(BlobId.class))).thenReturn(result);
+            doThrow(new StorageException(new IOException("Batch submit throws a storage exception"))).when(batch).submit();
+        } else {
+            StorageBatchResult<Boolean> resultA = mock(StorageBatchResult.class);
+            doReturn(resultA).when(batch).delete(eq(BlobId.of("bucket", "blobA")));
+            doAnswer(invocation -> {
+                StorageException storageException = new StorageException(new IOException("Batched delete throws a storage exception"));
+                ((BatchResult.Callback) invocation.getArguments()[0]).error(storageException);
+                return null;
+            }).when(resultA).notify(any(StorageBatchResult.Callback.class));
+
+            StorageBatchResult<Boolean> resultB = mock(StorageBatchResult.class);
+            doReturn(resultB).when(batch).delete(eq(BlobId.of("bucket", "blobB")));
+            doAnswer(invocation -> {
+                if (randomBoolean()) {
+                    StorageException storageException = new StorageException(new IOException("Batched delete throws a storage exception"));
+                    ((BatchResult.Callback) invocation.getArguments()[0]).error(storageException);
+                } else {
+                    ((BatchResult.Callback) invocation.getArguments()[0]).success(randomBoolean());
+                }
+                return null;
+            }).when(resultB).notify(any(StorageBatchResult.Callback.class));
+
+            doNothing().when(batch).submit();
+        }
+
+        final Storage storage = mock(Storage.class);
+        when(storage.get("bucket")).thenReturn(mock(Bucket.class));
+        when(storage.batch()).thenReturn(batch);
+
+        final GCSService storageService = mock(GCSService.class);
+        when(storageService.client(any(RepositoryMetadata.class))).thenReturn(storage);
+        RepositoryMetadata repositoryMetadata = new RepositoryMetadata("test", "bla", Settings.EMPTY);
+        try (BlobStore store = new GCSBlobStore("bucket", storageService, repositoryMetadata, randomIntBetween(1, 8) * 1024)) {
+            final BlobContainer container = store.blobContainer(new BlobPath());
+            assertThatThrownBy(() -> container.deleteBlobsIgnoringIfNotExists(blobs)).isInstanceOfAny(IOException.class).hasCauseExactlyInstanceOf(StorageException.class);
+        }
+    }
```

Comment:
This looks like all mocked errors are thrown by the GCS SDK and not in our layers, is this correct? Or what of our code is tested here?

Reply:
It is testing the error handling of GCSBlobStore.deleteBlobsIgnoringIfNotExists().

Reply:
https://github.com/crate/crate/pull/15598/files/b925e6894b9a3f4d2d50eec4d7d0da13bc88cf33#diff-f3e42da70c43a31f98a4eec4dd720092b684e72667effa01f4bf65f3b1939f36R309-R348

Reply:
The GCS SDK is throwing StorageExceptions, but they should not come out.

Reply:
The code is testing that all Exceptions from the gcs libs are handled accordingly and converted to IOExceptions.